### ISSUE TITLE
Windows 10 / Server 2016 Port

### ISSUE THREAD

**Issue by ramseur** [eveseat/seat#326]
Dear Seat Community,

I am sure you have already heard of the idea of running your application purely on Windows.  I was wondering if you had any plans to do so for those of us more familiar with windows.  You could also make a docker version for windows.

In theory, php and mysql can both easily run on Windows.  Youd only need port over your services and make them available to your application.  Then your application could run native on both linux and windows or in docker.  This would increase your adoption.

**Comment by warlof** [comment]
@ramseur o/

seems you don't put any eye in doc ;)
plus, constraint on Windows core is related to redis, our caching and queue system :p

requirements : https://eveseat.github.io/docs/general/requirements/
quick start : https://eveseat.github.io/docs/guides/installation/basic_installation/installation/
advanced : https://eveseat.github.io/docs/guides/installation/manual_installation/docker/

**Comment by ramseur** [comment]
Understood.  Ill try to use these and see if it works on windows.  They run redis on windows now I thought.

**Comment by ramseur** [comment]
I know how to install it on docker with linux.  I am more interested installing it on Windows Native since Redis or another cache system can easily be installed on windows. :)    We have a windows server and it would be nice to just host it plain on windows with redis installed.

**Comment by IvandaNothabeer** [comment]
Seat does run quite well on windows for a DEVELOPMENT environment. But without support for supervisord on windows the Seat team have (quite rightly) not provided support. Supervisord is essential for a production environment.

To run Seat on windows, follow the installation guides for Ubuntu, install the relevant components 

`composer --ignore-platform-reqs install eaveseat/seat`
and
`composer --ignore-platform-reqs update`

Should allow you to ignore the php requirements for running horizon.

To mimic the functions of supervisord, open a CMD window and run `php artisan queue:work`

But to repeat, without supervisord, running seat on windows as a production system is just not feasible.

**Comment by warlof** [comment]
I didn't get what was wrong with Docker and Linux container? I've no issue with it.

**Comment by warlof** [comment]
@ramseur try to restart seat-app container and nginx one.
I have this sometimes; but must say that I didn't pay much more attention to it since it's enough to dev on my main computer which is under Windows as well.

**Comment by ramseur** [comment]
@leonjza I need a slack invite sent to ramseur@anant.us

@warlof  Are you saying simply tell those two containers to restart in docker?  Odd because even when i change settings in both the compose and env file it still has that error.  People say its because the container its referring to itself.   Are you running windows 2016 or windows 10?  Did you have to set up a Nat network?  Mine was already setup when I upgraded to an Azure HyperV Server.  I think its something small.  Im assume I dont have to run some db script on fresh install and you guys run that in your app?  I also thought it could be the db user/password/database names but ive messed around with that to no luck.

Basically I am trying to accomplish the following things

- Run in Prod on Windows 2016 inside of a Hyper V Linux Container ( seems to be working, just a Ngix issue)
- Run Seat Native on Windows 2016 from suggestions @IvandaNothabeer and add some equiv of supervisord so that Seat can be supported native on both windows and linux.  Like I said, this would greatly increase adoption on the Windows side.

**Comment by warlof** [comment]
https://eveseat.github.io/docs/about/contact/

**Comment by ramseur** [comment]
I figured this out with the help of the team on Slack qu1ckkkk  and warlof 👍 

- Install Windows 10 or Windows 2016 with Hyper-V
- Download the yml and env file
- Rename the env file to local.env  ( or whatever you want but you have to use it later)
- Open up your local.env file
- Change your env file application key to a 32 length string
- Set debug = true just till you get it working
- Set your db user/database/password in your local.env file
- Open up your yml file
- Remove all the environmental variables and set them manually ( this is a pain until I figure out how to proper use the env file with compose in windows as it failed)
- Make sure you remove and set all the environmental vars in mariadb,nginx
- Change your "env_file:" entries to your "- ./local.env"
- docker-compose up as normal
- You should be able to be in windows but run a linux container . Open your IE or chrome browser to localhost:8080
- You can use IIS to redirect localhost:8080 to 80 if you want to host publicly on a domain.



Sample Docker Compose for Linux on windows

version: "3.2"

services:

  mariadb:
    image: mariadb:10.3
    container_name: seat-mariadb
    restart: always
    environment:
      MYSQL_RANDOM_ROOT_PASSWORD: "yes"
      MYSQL_USER: seat
      MYSQL_PASSWORD: seat
      MYSQL_DATABASE: seat
    volumes:
      - "mariadb-data:/var/lib/mysql"
    networks:
      - seat-network

  redis:
    image: redis:3
    container_name: seat-redis
    command: redis-server --appendonly yes
    restart: always
    volumes:
      - "redis-data:/data"
    networks:
      - seat-network

  nginx:
    image: eveseat/eveseat-nginx:latest
    depends_on:
      - seat-app
    container_name: seat-nginx
    restart: always
    environment:
      - NGINX_HOST=_
    volumes:
      - "seat-code:/var/www/seat"
      # Remove the comment below to enable nginx logs to disk.
      #- ./logs/nginx/:/var/log/nginx/
    ports:
      - "8080:80"
      - "8443:443"
    command: /bin/sh -c "envsubst '_' < /etc/nginx/nginx.conf.template > /etc/nginx/nginx.conf && nginx -g 'daemon off;'"
    networks:
      - seat-network

  seat-app:
    image: eveseat/eveseat-app:latest
    container_name: seat-app
    restart: always
    environment:
      DB_HOST: mariadb
      REDIS_HOST: redis
    env_file:
      - ./local.env
    volumes:
      - "seat-code:/var/www/seat"
      # Remove the comment below to enable SeAT logs to disk.
      #- ./logs:/var/www/seat/storage/logs
    depends_on:
      - mariadb
      - redis
    networks:
      - seat-network

  seat-worker:
    image: eveseat/eveseat-worker:latest
    container_name: seat-worker
    restart: always
    environment:
      DB_HOST: mariadb
      REDIS_HOST: redis
    env_file:
      - ./local.env
    volumes:
      - "seat-code:/var/www/seat"
    depends_on:
      - seat-app # for the seat-code volume
      - mariadb
      - redis
    networks:
      - seat-network

  seat-cron:
    image: eveseat/eveseat-cron:latest
    container_name: seat-cron
    restart: always
    environment:
      DB_HOST: mariadb
      REDIS_HOST: redis
    env_file:
      - ./local.env
    volumes:
      - "seat-code:/var/www/seat"
    depends_on:
      - seat-app # for the seat-code volume
      - mariadb
      - redis
    networks:
      - seat-network

volumes:
    seat-code:
    redis-data:
    mariadb-data:

networks:
    seat-network:


sample local.env file 


APP_DEBUG=true


APP_URL=http://localhost


EVE_CLIENT_ID=null
EVE_CLIENT_SECRET=null
EVE_CALLBACK_URL=https://seat.yourdomain.com/auth/eve/callback

APP_KEY=gsdsertyuikjhngbvfdcxsdwertghjyt



DB_DATABASE=seat
DB_USERNAME=seat
DB_PASSWORD=seat


MAIL_DRIVER=smtp
MAIL_HOST=smtp.mailtrap.io
MAIL_PORT=2525
MAIL_USERNAME=null
MAIL_PASSWORD=null
MAIL_ENCRYPTION=null
MAIL_FROM_ADDRESS=noreply@localhost.local
MAIL_FROM_NAME='SeAT Administrator'




NGINX_HTTP=8080
NGINX_HTTPS=8443


NGINX_HOST=_

**Comment by leonjza** [comment]
You're welcome and thanks for the feedback!

**Comment by warlof** [comment]
so, is it close ?
